src/content/docs/en/guides/upgrade-to/v4.mdx

```diff
@@ -0,0 +1,387 @@
+  <Fragment slot="yarn">
+  ```shell
+  # Upgrade Astro and official integrations together
+  yarn dlx @astrojs/upgrade
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+
+You can also [upgrade your Astro integrations manually](/en/guides/integrations-guide/#manual-upgrading) if needed, and you may also need to upgrade other dependencies in your project.
+
+
+:::note[Need to continue?]
+After upgrading Astro to the latest version, you may not need to make any changes to your project at all!
+
+But, if you notice errors or unexpected behavior, please check below for what has changed that might need updating in your project.
+:::
+
+Astro v4.0 includes [potentially breaking changes](/en/guides/upgrade-to/v4/#breaking-changes), as well as the [removal of some previously deprecated features](/en/guides/upgrade-to/v4/#previously-deprecated-features-now-removed). 
+
+If your project doesn't work as expected after upgrading to v4.0, check this guide for an overview of all breaking changes and instructions on how to update your codebase.
+
+See [the changelog](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md) for full release notes.
+
+## Astro v4.0 Experimental Flags Removed
+
+Remove the following experimental flags from `astro.config.mjs`:
+
+```js del={6}
+// astro.config.mjs
+import { defineConfig } from 'astro/config';
+
+export default defineConfig({
+  experimental: {
+    // nothing here yet @TODO: check if needed
+  },
+})
+```
+
+These features are now available by default:
+
+- 
+- 
+
+Read more about these two exciting features and more in [the v4.0 Blog post](https://astro.build/blog/astro-4/)!
+
+## Upgrades
+
+Any major upgrades to Astro's dependencies may cause breaking changes in your project.
+
+### Upgraded: Vite 5.0
+
+In Astro v3.0, Vite 4 was used as the development server and production bundler.
+
+Astro v4.0 upgrades from Vite 4 to Vite 5.
+
+#### What should I do?
+
+If you are using Vite-specific plugins, configuration or APIs, check the [Vite migration guide](https://vitejs.dev/guide/migration) for their breaking changes and upgrade your project as needed. There are no breaking changes to Astro itself.
+
+### Upgraded: unified, remark, and rehype dependencies
+
+In Astro v3.x, unified v10 and its related compatible remark/rehype packages were used to process Markdown and MDX.
+
+Astro v4.0 upgrades [unified to v11](https://github.com/unifiedjs/unified/releases/tag/11.0.0) and the other remark/rehype packages to latest.
+
+#### What should I do?
+
+If you used custom remark/rehype packages, update all of them to latest using your package manager to ensure they support unified v11. The packages you are using can be found in `astro.config.mjs`.
+
+There should not be any significant breaking changes if you use actively-updated packages, but some packages may not yet be compatible with unified v11.
+Visually inspect your Markdown/MDX pages before deploying to ensure your site is functioning as intended.
```

Comment:
I understand `latest` here means the `@latest` suffix on package managers, but since this might not be something everyone's aware of, they might think the writing structure is wrong, and also, this makes it harder for translations to adapt. I would just use "to the latest version".

Also, we have a hyphen that shouldn't exist!
```suggestion
Astro v4.0 upgrades [unified to v11](https://github.com/unifiedjs/unified/releases/tag/11.0.0) and the other remark/rehype packages to the latest version.

#### What should I do?

If you used custom remark/rehype packages, update all of them to the latest version using your package manager to ensure they support unified v11. The packages you are using can be found in `astro.config.mjs`.

There should not be any significant breaking changes if you use actively updated packages, but some packages may not yet be compatible with unified v11.
Visually inspect your Markdown/MDX pages before deploying to ensure your site is functioning as intended.
```